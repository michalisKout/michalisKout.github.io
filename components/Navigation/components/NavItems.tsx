import { FC } from 'react';
import Link from 'next/link';
import NavItem from './NavItem';

interface Props {
  wrapperClassName?: string;
  itemClassName?: string;
}

const menuConfig = [
  {
    title: 'Linkedin',
    url: 'https://www.linkedin.com/in/michalis-koutridis-340774124/',
  },
  {
    title: 'Github',
    url: 'https://github.com/michalisKout',
  },
  {
    title: 'Twitter',
    url: 'https://twitter.com/Michael_Kout',
  },
];

const NavItems: FC<Props> = ({
  wrapperClassName = 'nav-items--wrapper',
  itemClassName = 'nav-item',
}) => (
  <div className={`${wrapperClassName}`}>
    {menuConfig.map(({ url, title, index }) => (
      <NavItem key={`${title}_${index}`} url={url} title={title} itemClassName={itemClassName} />
    ))}
  </div>
);

export default NavItems;
