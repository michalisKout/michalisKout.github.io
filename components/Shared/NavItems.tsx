import { FC } from "react";
import Link from "next/link";

interface Props {
  wrapperClassName?: string;
  itemClassName?: string;
}
const NavItems: FC<Props> = ({
  wrapperClassName = "nav-items--wrapper",
  itemClassName = "nav-item",
}) => (
  <div className={`${wrapperClassName}`}>
    <div className={`${itemClassName}`}>
      <Link href={"/"}>Home</Link>
    </div>
    <div className={`${itemClassName}`}>
      <Link href={"/portfolio"}>Portfolio</Link>
    </div>
    <div className={`${itemClassName}`}>
      <Link href={"/experience"}>Experience</Link>
    </div>
    <div className={`${itemClassName}`}>
      <Link href={"/contact"}>Contact</Link>
    </div>
  </div>
);

export default NavItems;
