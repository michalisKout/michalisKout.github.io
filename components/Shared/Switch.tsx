import React from "react";
import useToggle from "core/hooks/utils/useToggle";

const Switch = () => {
  const { toggle, handleClick } = useToggle();
  return (
    <div className="switch" onClick={handleClick}>
      <input type="checkbox" checked={toggle} readOnly />
      <span className="slider" />
    </div>
  );
};

export default Switch;
