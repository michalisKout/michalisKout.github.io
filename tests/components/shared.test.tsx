import { render } from '@testing-library/react';
import NavItems from '../../components/Navigation/components/NavItems';
import Switch from 'components/Shared/Switch';

describe('Shared Components', () => {
  it('should render all the nav items', () => {
    const { getAllByTestId } = render(<NavItems />);
    const navItems = getAllByTestId(/nav-item/);

    expect(navItems.length).toBe(3);
  });

  it('should render Switch component', () => {
    const { asFragment } = render(<Switch />);

    expect(asFragment()).toMatchSnapshot();
  });
});
