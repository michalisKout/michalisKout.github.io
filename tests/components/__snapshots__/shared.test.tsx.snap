// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Components should render Switch component 1`] = `
<DocumentFragment>
  <div
    class="switch"
    data-testid="switch-button"
  >
    <div
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(31, 41, 55); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
      >
        <div
          style="height: 28px; width: 30px; position: relative; opacity: 0; pointer-events: none; transition: opacity 0.25s;"
        >
          <test />
        </div>
        <div
          style="height: 28px; width: 30px; position: absolute; opacity: 1; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
        >
          <test />
        </div>
      </div>
      <div
        class="react-switch-handle"
        style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        aria-checked="false"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </div>
</DocumentFragment>
`;
